device/protocol_mqtt_v1.go

```diff
@@ -438,6 +497,9 @@ func (d *Device) sendEmptyCache() error {
 	// Set up empty cache
 	emptyCacheTopic := fmt.Sprintf("%s/control/emptyCache", d.getBaseTopic())
 
+	// Erase our server-owned properties in the storage (we want a fresh start)
+	d.removeAllServerOwnedPropertiesFromStorage()
```

Comment:
Here we might want some more coherence across SDKs. For example, the Rust SDK [cleans up the whole property db](https://github.com/astarte-platform/astarte-device-sdk-rust/blob/fc7366324768bfa61a9b9c9e68a62e72a7a73fdb/src/lib.rs#L148-L150) after publishing the emptycache message, while the [Java](https://github.com/astarte-platform/astarte-device-sdk-java/blob/3de54b55d6074d62e468c2cef7030f791426fae3/DeviceSDK/src/main/java/org/astarteplatform/devicesdk/transport/mqtt/AstarteMqttV1Transport.java#L471) and [Qt5](https://github.com/astarte-platform/astarte-device-sdk-qt5/blob/73b76bbb22e2d0d2422e94461e843fc01b3af22b/astarte-transport/astartetransport.cpp#L522) SDKs seem to do nothing regarding storage. Maybe track this in a meta-issue?

Reply:
Yes, but clearing the whole property DB seems like the wrongest behavior possible and I think this is a huge bug in the Rust SDK. Empty cache doesn't mean that information is invalid - it means that each end doesn't trust another anymore, and as such it should re-receive all information. Suppose e.g. a property was deleted before the empty cache, then empty cache is called - if you don't erase the DB, the client will still consider that property as valid. On the other hand, if you erase the whole database AND do an empty cache, when resending the properties to the server you will end up sending nothing. So I think this one is indeed the correct behaviour, and we should probably take action in other SDKs.

Reply:
After a private discussion, this has been handled by implementing control messages